src/test/kotlin/uk/gov/dluhc/registercheckerapi/service/RegisterCheckServiceTest.kt

```diff
@@ -329,7 +369,7 @@ internal class RegisterCheckServiceTest {
                 "11, TOO_MANY_MATCHES, TOO_MINUS_MANY_MINUS_MATCHES"
             ]
         )
-        fun `should update pending register check successfully and submit a ConfirmRegisterCheckResult Message`(
+        fun `should update pending register check successfully`(
```

Comment:
Is there an easy way to test, that the message is only sent when there is no a locking exception. Many in the service test you can mock that the exception is thrown.

Reply:
so these are now separated into separate functions in the service test and it is only at the controller level that the optimistic locking is handled (because it applies to the whole transaction). 

So that test case is covered by the integration test as it doesn't make sense on the service level

Reply:
Thanks that makes sense. Let's leave it then.